rules/dead-doc-block/src/TagRemover/VarTagRemover.php

```diff
@@ -104,6 +104,19 @@ private function isNonBasicArrayType(Node $node, VarTagValueNode $varTagValueNod
         return $varTypeDocString !== 'array';
     }
 
+    private function doesClassOrInterfaceExists(AttributeAwareArrayTypeNode $type): bool
+    {
+        if (class_exists((string) $type->type)) {
+            return true;
+        }
+
+        if (interface_exists((string) $type->type)) {
+            return true;
+        }
+
+        return false;
+    }
+
```

Comment:
Good catch!

There is `Symplify\PackageBuilder\Reflection\ClassLikeExistenceChecker` service that can be re-used. It will help us to catch all existance edge cases without maintenance.

Reply:
Great, added in https://github.com/rectorphp/rector/pull/5377/commits/32abc0b37dd69eb46050952e7c3d9f11b6c9d653